client/src/main/java/org/opensearch/ml/client/MachineLearningClient.java

```diff
@@ -460,7 +460,23 @@ default ActionFuture<DeleteResponse> deleteAgent(String agentId) {
         return actionFuture;
     }
 
-    void deleteAgent(String agentId, ActionListener<DeleteResponse> listener);
+    /**
+     * Delete agent
+     * @param agentId The id of the agent to delete
+     * @param listener a listener to be notified of the result
+     */
+    default void deleteAgent(String agentId, ActionListener<DeleteResponse> listener) {
+        PlainActionFuture<DeleteResponse> actionFuture = PlainActionFuture.newFuture();
+        deleteAgent(agentId, null, actionFuture);
+    }
+
+    /**
+     * Delete agent
+     * @param agentId The id of the agent to delete
+     * @param tenantId the tenant id. This is necessary for multi-tenancy.
+     * @param listener a listener to be notified of the result
+     */
+    void deleteAgent(String agentId, String tenantId, ActionListener<DeleteResponse> listener);
```

Comment:
Do we need to expose transport API for delete agent for multi-tenancy ?

Reply:
Ideally request should land on Flow framework and flow framework will invoke this transport api.